types/src/transaction/authenticator.rs

```diff
@@ -43,7 +44,7 @@ pub enum AuthenticationError {
 pub enum TransactionAuthenticator {
     /// Single signature
     Ed25519 {
-        public_key: Ed25519PublicKey,
+        public_key: Validatable<Ed25519PublicKey>,
         signature: Ed25519Signature,
```

Comment:
We don't apply `Validatable` for signature?

Reply:
We could, although after talking with @zekun000 (and looking at the code) validation for Signatures is incredibly cheap compared to validation for Public Keys

Reply:
Actually the main question is if the rest of Authenticators (multi-sig and multi-agent atm, but ECDSA has been proposed too) would require that handling as well. Ideally, we want some consistent policy here, but it's a good phase1 step.

Reply:
Yeah it wouldn't be much more work to do something similar for the others but felt starting with one case would be a good first step.